contracts/root/RootChain.sol

```diff
@@ -57,7 +57,7 @@ contract RootChain is RootChainStorage {
     onlyDepositManager
     returns(uint256 depositId)
   {
-    depositId = _blockDepositId;
+    depositId = _nextHeaderBlock.sub(MAX_DEPOSITS).add(_blockDepositId);
```

Comment:
Please explain.

Reply:
If 1 header block was submitted, `_nextHeaderBlock` is 20000.
So `depositId = 20000 - 10000 + 1` where `_blockDepositId` is initialized to 1 at every checkpoint.

If no header blocks have been submitted, `_nextHeaderBlock` is 10000.
So `depositId = 10000 - 10000 + 1`

Basically `_blockDepositId` denotes  deposit count relative to checkpoint and  `depositId = _nextHeaderBlock.sub(MAX_DEPOSITS).add(_blockDepositId)` evaluates absolute depositId.